sdk/cognitiveservices/azure-cognitiveservices-formrecognizer/DESIGN_DOC.md

```diff
@@ -0,0 +1,544 @@
+
+class PageMetadata:
+    page_number: int
+    angle: float
+    width: int
+    height: int
+    unit: str
+    language: str
+    lines: List[ExtractedLine]
+```
+
+### Layout Sample
+
+```python
+import pandas as pd
+from azure.ai.formrecognizer import FormRecognizerClient
+
+client = FormRecognizerClient(endpoint, credential)
+
+table_image = "https://www.traveldailymedia.com/assets/2020/01/Table2.png"
+poller = client.begin_extract_layout(table_image)
+result = poller.result()
+
+table = result[0].tables[0] # page 1, table 1
+my_table = [[None for x in range(table.column_count)] for y in range(table.row_count)]
+for cell in table.cells:
+    my_table[cell.row_index][cell.column_index] = cell.text
+
+dftable = pd.DataFrame(my_table)
+print(dftable.to_markdown())
+```
+
+## Custom
+
+The `CustomFormClient` provides all the operations necessary for training a custom model, analyzing a form with a 
+custom model, and managing a user's custom models on their account.
+
+The user can choose to train with or without labels using the methods `begin_labeled_training` or `begin_training`. 
+Both methods take as input a blob SAS uri or valid path to the documents to use for training. Each training method 
+will return a poller object which is used to get the training result.
+
+A custom model can be used to analyze forms using the `begin_extract_forms` or `begin_extract_labeled_forms` method.
+The `model_id` from the training result is passed into the methods, along with the input form to analyze (content-type
+is determined internally). This method also returns a poller object which is used to get the result.
+
+In order for the user to manage their custom models, a few methods are available to list custom models, delete a model,
+get a models summary for the account, or get a custom model.
+
+### Custom Form Client
+```python
+from azure.ai.formrecognizer import CustomFormClient
+
+client = CustomFormClient(endpoint: str, credential: Union[CognitiveKeyCredential, TokenCredential])
+
+# Train
+client.begin_labeled_training(
+    source: str, source_prefix_filter: str, include_sub_folders: bool=False
+) -> LROPoller -> CustomLabeledModel
+
+client.begin_training(
+    source: str, source_prefix_filter: str, include_sub_folders: bool=False
+) -> LROPoller -> CustomModel
+
+# Extract
+client.begin_extract_forms(form: Union[str, BytesIO], model_id: str) -> LROPoller -> List[ExtractedPage]
+
+client.begin_extract_labeled_forms(form: Union[str, BytesIO], model_id: str) -> LROPoller -> ExtractedForm
+
+# Manage
+client.list_custom_models() -> ItemPaged[ModelInfo]
+
+client.get_models_summary() -> ModelsSummary
+
+client.get_custom_model(model_id: str) -> CustomModel
+
+client.get_custom_labeled_model(model_id: str) -> CustomLabeledModel
+
+client.delete_custom_model(model_id: str) -> None
+```
+
+### Custom Models Unlabeled
+```python
+# Training ---------------------------------------------------
+class CustomModel:
+    model_id: str
+    status: str
```

Comment:
Should we use a python enum here (and other status strings)? An alternative might be to expose `bool` properties `succeeded`, `partially_succeeded`, `failed` directly on the object.

Reply:
Sure, added enums `ModelStatus` and `TrainStatus`